src/server/pkg/storage/chunk/storage.go

```diff
@@ -39,7 +44,7 @@ func (s *Storage) NewReader(ctx context.Context, dataRefs ...*DataRef) *Reader {
 // Chunks are created based on the content, then hashed and deduplicated/uploaded to
 // object storage.
 func (s *Storage) NewWriter(ctx context.Context, tmpID string, f WriterFunc, opts ...WriterOption) *Writer {
-	return newWriter(ctx, s.objClient, s.gcClient, tmpID, f, opts...)
+	return newWriter(ctx, s.objClient, s.gcClient, tmpID, f, append(opts, WithChunkTTL(defaultChunkTTL))...)
```

Comment:
I think the order is wrong here. You want the default option first, then the client supplied options afterwards.